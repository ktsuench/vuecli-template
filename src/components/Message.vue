<template>
  <div>{{ msg }}</div>
</template>

<script>
export default {
  name: 'message',
  props: {
    msg: {
      type: String,
      default: 'default message',
    },
  },
};
</script>
