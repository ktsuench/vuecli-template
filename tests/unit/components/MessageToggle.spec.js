import { shallowMount } from '@vue/test-utils';
import MessageToggle from '@/components/MessageToggle.vue';
import Message from '@/components/Message.vue';

describe('MessageToggle.vue', () => {
  it('toggles msg passed to Message when button is clicked', () => {
    const wrapper = shallowMount(MessageToggle);
    const button = wrapper.find('#toggle-message');
    const MessageComponent = wrapper.find(Message);

    button.trigger('click');
    expect(MessageComponent.props()).toEqual({ msg: 'message' });

    button.trigger('click');
    expect(MessageComponent.props()).toEqual({ msg: 'toggled message' });
  });

  it('matches snapshot', () => {
    const wrapper = shallowMount(MessageToggle);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
