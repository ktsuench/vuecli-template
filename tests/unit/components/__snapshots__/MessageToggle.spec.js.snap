// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageToggle.vue matches snapshot 1`] = `
<div>
  <message-stub></message-stub> <button id="toggle-message">
    Change message
  </button>
</div>
`;
